test-lib.sh

```diff
@@ -195,7 +213,30 @@ function ct_show_results() {
 # --------------------
 # Enables automatic container cleanup after tests.
 function ct_enable_cleanup() {
-  trap ct_cleanup EXIT SIGINT
+  trap ct_trap_on_exit EXIT
+  trap ct_trap_on_sigint SIGINT
+}
+
+# ct_trap_on_exit
+# --------------------
+function ct_trap_on_exit() {
+  local exit_code=$?
+  [ $exit_code -eq 130 ] && return # we do not want to catch SIGINT here
+  echo "Tests finished with EXIT=$exit_code"
+  exit_code=$(("${TESTSUITE_RESULT:-0}"+exit_code))
```

Comment:
We should not really care about what the script returns, as now all the relevant `exit` calls are removed from the tests.

I put it there for future, just to be sure that we catch some not expected behavior, if any is added.

There should not be any additional problems with this line as I see it. But sure, I can add some explanatory comment. Or remove the functionality at all, if you think it is unnecessary.

Reply:
I think we can keep it as is but a comment would be nice. The only thing I would be afraid of is the availability of this construct in some earlier versions of shells, since we have seen some operating systems (especially Mac) to be stuck in the past. :) But we can come back to this if we get complaints.

Reply:
Actually, this problem already stroke on CentOS7. So I need to rework it. Thanks for catching it :).

Reply:
Fixed with: https://github.com/sclorg/container-common-scripts/pull/338/commits/c0499b7b040b23fc004cfd90bd2f02be7d079d82.